src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -333,97 +233,460 @@ struct jit_uni_mvn_mean_variance_kernel_f32 : public jit_uni_mvn_mean_variance_k
+        // safe to use abi reg now.
+        Xbyak::Reg64 reg_src_bk = rcx;
+        Xbyak::Reg64 reg_work_amount_bk = rdi;
+        mov(reg_src_bk, reg_src);
+        mov(reg_work_amount_bk, reg_work_amount);
+        int repeats = (isa == cpu::x64::sse41) ? 2 : 1; // block size is also 8 on cpu::x64::sse41 with two step process
+
+        auto reset_with_offset = [&](int offset) {
+            add(reg_src_bk, offset * jcp_.src_data_size);
+            mov(reg_src, reg_src_bk);
+            mov(reg_work_amount, reg_work_amount_bk);
+            if (jcp_.normalize_variance) {
+                // mean and vaiance for variance kernel
+                if (!jcp_.across_channels) {
+                    // mean is bc when across_channel, no need shift
+                    add(reg_mean, offset * sizeof(float));
+                    uni_vmovups(vmm_mean, ptr[reg_mean]);
+                }
+                add(reg_variance, offset * sizeof(float));
+                uni_vpxor(vmm_variance, vmm_variance, vmm_variance);
+            } else {
+                // sum for mean kernel
+                add(reg_sum, offset * sizeof(float));
+                uni_vpxor(vmm_sum, vmm_sum, vmm_sum);
+            }
+        };
+
+        auto save_result = [&]() {
+            // add input_base value and store for per_channel
+            // store for across_channels
+            if (jcp_.normalize_variance) {
+                if (!jcp_.across_channels) {
+                    uni_vmovups(vmm_val, ptr[reg_variance]);
+                    uni_vaddps(vmm_variance, vmm_variance, vmm_val);
+                }
+                uni_vmovups(ptr[reg_variance], vmm_variance);
+            } else {
+                if (!isFloatCompatible(jcp_.src_prc))  // add with int for int-family data type, other compute go with float
+                    uni_vcvtdq2ps(vmm_sum, vmm_sum);
+
+                if (!jcp_.across_channels) {
+                    uni_vmovups(vmm_val, ptr[reg_sum]);
+                    uni_vaddps(vmm_sum, vmm_sum, vmm_val);
+                }
+                uni_vmovups(ptr[reg_sum], vmm_sum);
+            }
+        };
+
+        auto worker_tails_unroll = [&]() {
+            auto unroll_w = [&](int block_num) {
+                Xbyak::Label loop_label;
+                Xbyak::Label loop_end_label;
+                L(loop_label);
+                {
+                    cmp(reg_work_amount, 0);
+                    jle(loop_end_label, T_NEAR);
+
+                    worker_block(block_num, true);
+
+                    add(reg_src, src_stride);
+                    sub(reg_work_amount, 1);
+
+                    jmp(loop_label, T_NEAR);
+                }
+                L(loop_end_label);
+            };
+            Label tail_blk8_label;
+            Label tail_blk8_exit_label;
+            Label tail_blk4_label;
+            Label tail_blk4_exit_label;
+            Label tail_blk1_label;
+            Label tail_blk1_exit_label;
+            L(tail_blk8_label);
```

Comment:
The idea is to avoid code duplication by wrapping tiles processing into `for` loop. Look smt like that: 
```
Label tail_start[3]
Label tail_exit[3]
int[3] tails_sizes = {8, 4, 1}

for (int i = 0; i < 3; i++)
    L(tail_start[i]);
    {
        cmp(reg_rt_shape, tails_sizes[i]);
        jl(tail_exit[i], T_NEAR);

        unroll_w(tails_sizes[i]);
        save_result();
        reset_with_offset(tails_sizes[i]);

        sub(reg_rt_shape, tails_sizes[i]);
        jmp(tail_start[i], T_NEAR);
    }
    L(tail_start[i]);
```
Applying such approach for the whole MVN you can achive great adjustability of the tile processing. Like you can have one array of desired tile sizes which will be automatiacally applied the each part of the impl.

Reply:
Applied anywhere applicable, the single tail function with any work_type as parameters(std::function). (except nspc&&per_channel case, which is a dedicated process with unroll similar vector part, unify it seem don't make more clear), Welcome comment if you have more concerns. Thanks Dmitry. [d464ca6](https://github.com/openvinotoolkit/openvino/pull/17988/commits/d464ca6a2cb92f858670f01553892b975c0c20f5)